core/trait_card.py

```diff
@@ -18,3 +21,9 @@ def __eq__(self, other: object) -> bool:
     def __hash__(self) -> int:
         return hash(tuple(vars(self).values()))
 
+    def set_owner(self, owner: object):
+        self._owner = weakref.ref(owner)
+
```

Comment:
Suggest adding a comment to explain what owner is or renaming owner or both. It took me a while to understand why we are calling self._owner in long_neck.py.

Reply:
I'll rename it to parent_species